File: environment/pset6/caesar/caesar.py
from cs50 import get_string
from cs50 import sys
from sys import argv


sys.argv[1]
k=int(sys.argv[1])
if(len(argv))==2:


    s=get_string("plaintext:")
    print("ciphertext:",end="")


    for c in s:




        if c.islower():

            n=ord (c)
            o=n-97
            r=(o+k)%26
            i=r+97
            u=chr(i)
            print(u,end="")


        if c.isupper():

            g=ord (c)
            l=g-65
            j=(l+k)%26
            d=j+65
            w=chr(d)
            print(w,end="")

        if (c==","):

            print(",",end="")

        if (c==" "):

            print(" ",end="")

        if (c=="!"):

            print("!",end="")




    print()
else:
    print("retry")
    exit(2)

File: environment/whodunit/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
   string s;
   {
   if (argc==2)
   {
     string num = argv[1];
     int i= atoi (num);
     s=get_string();
     for (int p=0, n=strlen(s); p<n; p++)
     {
        if(islower(s[p]))
        {
           int x= s[p] -97;
           int f=(x+i)%26;
           int y=f+97;
           printf("%c",y);
        }
        else if(isupper(s[p]))
        {
           int g= s[p] -65;
           int u=(g+i)%26;
           int k=u+65;

           printf("%c",k);

        }

     }
     printf("\n");

}

   else
{
     printf("false\n");
}
}
}